### PR TITLE
Fix/theme flicker

### LINKED ISSUE #1711: dark mode pages flicker with white background in iframes
see: https://cal.com/peerrich-eth/orange-dao or  https://cal.com/web3

https://user-images.githubusercontent.com/8019099/152646668-a81c1f40-10a3-4624-a04e-badf006c1f38.mov

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

This is a fix for flicker of Light when using a dark theme. The current approach of useLayoutEffect doesn't solve the problem with slow connections. The issue is more easily visible on https://cal.com/web3 even with good connections

**With useLayoutEffect(Current Approach)**
with-layoutEffect

**Suggested Change**
suggested-change

Fixes #1711.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [x] I checked the refresh behaviour of Availability, EventType pages  with Dark Theme

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code and corrected any misspellings
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by hariombalhara** [comment]
@PeerRich  As per your comment on other Pull Request 
"can you two make sure that the logged in dashboard is always in light mode? right now we only have light/dark mode for the public booking pages"
yes, the dashboard will be in light mode only as the change would effect the places where useTheme was in use. I checked manually also.

File: components/booking/pages/AvailabilityPage.tsx [hunk 1]
```diff
@@ -99,6 +99,7 @@ const AvailabilityPage = ({ profile, eventType, workingHours }: Props) => {
 
   return (
     <>
+      <Theme />
```
Comment (hariombalhara) [inline components/booking/pages/AvailabilityPage.tsx:102]: @PeerRich Fixed the Theme component usage.

**Comment by hariombalhara** [comment]
@zomars Regarding this comment

> I can see that this works only on refresh. Can we make this work so we listen for that media query instead and changes display live?

Would it be okay if I do it in another Pull Request as this would be an enhancement.

**Comment by zomars** [comment]
> Would it be okay if I do it in another Pull Request as this would be an enhancement.

That would be awesome @hariombalhara

**Comment by PeerRich** [comment]
whats missing to be merged?

**Review by PeerRich — APPROVED** [review]
not the cleanest solution but good for now! thank you for the contribution

**Comment by hariombalhara** [comment]
> not the cleanest solution but good for now! thank you for the contribution

Yeah I agree. But couldn't find the best way to avoid the flash. The best solution seems to be adding the JS code in the head directly.

I have made some changes that would allow the JS string to be uglified and type checked. Will push them soon.

Do you have any suggestions on improving it otherwise?

**Comment by hariombalhara** [comment]
> > Would it be okay if I do it in another Pull Request as this would be an enhancement.
> 
> That would be awesome @hariombalhara

This is done in https://github.com/calcom/cal.com/pull/1807

**Pull request merged**